From 3d11e3724ffb3dc432c51736084aee081c660e0a Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 16 Dec 2015 13:16:39 +0800
Subject: [PATCH 0782/5242] MLK-12027: ARM: dts: fix the sample rate clock not
 accurate for spdif tx

commit  157640ee024c37f5b9102ef65a19f85aa0326e5a from
https://source.codeaurora.org/external/imx/linux-imx.git

Clock of spdif tx is derived from clk_ipg and clk_osc, which is not the
integer multiple size of sample rate, can't generate accurate clock for
each sample rate. Use pll4 as the clk_spdif's parent, because the clk_spdif
is the one of source clock of tx, use a proper frequency for pll4, then it
can generate more accurate clock for sample rate (32k,48k,96k,192k).

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-spdif.dts |    4 ++++
 arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts       |    5 +++++
 2 files changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-spdif.dts b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-spdif.dts
index ee53073..7191f05 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-spdif.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-spdif.dts
@@ -34,5 +34,9 @@
 &spdif {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_spdif>;
+	assigned-clocks = <&clks IMX6UL_CLK_SPDIF_SEL>,
+			  <&clks IMX6UL_CLK_SPDIF_PODF>;
+	assigned-clock-parents = <&clks IMX6UL_CLK_PLL4_AUDIO_DIV>;
+	assigned-clock-rates = <0>, <98304000>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts
index d31bd3c..1257352 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts
@@ -81,6 +81,11 @@
 	};
 };
 
+&clks {
+	assigned-clocks = <&clks IMX6UL_CLK_PLL4_AUDIO_DIV>;
+	assigned-clock-rates = <786432000>;
+};
+
 &cpu0 {
 	/*
 	 * on i.MX6UL, no seperated VDD_ARM_IN and VDD_SOC_IN,
-- 
1.7.9.5

